causal-analysis/TowardsCausalThinking.ipynb

```diff
@@ -0,0 +1,2117 @@
+       "<polygon fill=\"black\" stroke=\"black\" points=\"265.24,-43.28 270.84,-34.29 260.85,-37.83 265.24,-43.28\"/>\n",
+       "</g>\n",
+       "<!-- ConcurrentRules -->\n",
+       "<g id=\"node15\" class=\"node\">\n",
+       "<title>ConcurrentRules</title>\n",
+       "<ellipse fill=\"none\" stroke=\"black\" cx=\"376.84\" cy=\"-90\" rx=\"73.36\" ry=\"18\"/>\n",
+       "<text text-anchor=\"middle\" x=\"376.84\" y=\"-84.95\" font-family=\"Times,serif\" font-size=\"14.00\">ConcurrentRules</text>\n",
+       "</g>\n",
+       "<!-- ConcurrentRules&#45;&gt;Ruler -->\n",
+       "<g id=\"edge14\" class=\"edge\">\n",
+       "<title>ConcurrentRules&#45;&gt;Ruler</title>\n",
+       "<path fill=\"none\" stroke=\"black\" d=\"M356.22,-72.41C344.33,-62.85 329.26,-50.72 316.48,-40.44\"/>\n",
+       "<polygon fill=\"black\" stroke=\"black\" points=\"318.82,-37.83 308.84,-34.29 314.44,-43.28 318.82,-37.83\"/>\n",
+       "</g>\n",
+       "</g>\n",
+       "</svg>\n"
+      ],
+      "text/plain": [
+       "<graphviz.graphs.Digraph at 0x1140f55d0>"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "acmobsWritePathm=graphviz.Digraph()\n",
+    "acmobsWritePathm.edge('ObsAPIGW','metricCollector')\n",
+    "acmobsWritePathm.edge('Receiver','ObsAPIGW')\n",
+    "acmobsWritePathm.edge('ObjStore','Receiver')\n",
+    "acmobsWritePathm.edge('ReceiverPV','Receiver')\n",
+    "#acmobsWritePathm.edge('NumManagedClusters','Receiver')\n",
+    "acmobsWritePathm.edge('TotalTimeSeries','Receiver')\n",
+    "acmobsWritePathm.edge('NumReceiverReplica','Receiver')\n",
+    "\n",
+    "acmobsWritePathm.edge('Receiver','Querier')\n",
+    "\n",
+    "acmobsWritePathm.edge('TimeSeriesPerQuery','Querier')\n",
+    "acmobsWritePathm.edge('ConcurrentQuery','Querier')\n",
+    "acmobsWritePathm.edge('NumQueryReplica','Querier')\n",
+    "\n",
+    "acmobsWritePathm.edge('Querier','Ruler')\n",
+    "\n",
+    "acmobsWritePathm.edge('NumRules','Ruler')\n",
+    "acmobsWritePathm.edge('TimeSeriesPerRule','Ruler')\n",
+    "acmobsWritePathm.edge('ConcurrentRules','Ruler')\n",
+    "\n",
+    "\n",
+    "acmobsWritePathm"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9340c46d",
+   "metadata": {},
+   "source": [
+    "### GRC\n",
+    "\n",
+    "ACM GRC has many components -\n",
+    "\n",
+    "1. Policy Propagator on the Hub\n",
+    "1. Spec Sync Controller on the Managed cluster\n",
+    "1. Status Sync Controller on the Managed cluster\n",
+    "1. Policy Controllers on the managed cluster\n",
+    "1. Kube API Server on the Hub\n",
+    "1. Kube API Server on the managed cluster\n",
+    "\n",
+    "However, we must consider the health in lenses of what a user of the system sees. The user sees the effect when they try to `Create a Policy` or during/for `Policy Status Updates`.\n",
```

Comment:
It's really the creation (or update) of a replicated policy.  If there's no placement details -- a created policy is basically disabled.  As managed clusters are added or removed from the placement, this replicated policy creation is happening.